src/main/java/sx/blah/discord/api/internal/DiscordVoiceWS.java

```diff
@@ -257,11 +257,13 @@ public void run() {
 	 * Sets up the thread for receiving audio packets from the Discord UDP Socket.
 	 */
 	private void setupReceiveThread() {
-		Runnable receiveThread = () -> {
+		(new Thread((Runnable) () -> {
+                    while(true)
 			if (isConnected.get()) {
 				DatagramPacket receivedPacket = new DatagramPacket(new byte[1920], 1920);
 				try {
 					udpSocket.receive(receivedPacket); // This blocks the thread until a packet is received.
```

Comment:
good advice, I'll use isConnected, and SocketTimeout to make the thread shutdown properly when disconnected.